File: frontend/src/app/controller/service/admin/commun/PaiementIrAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';
import {environment} from 'src/environments/environment';
import {PaiementIrDto} from 'src/app/controller/model/commun/PaiementIr.model';
import {PaiementIrCriteria} from 'src/app/controller/criteria/commun/PaiementIrCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class PaiementIrAdminService extends AbstractService<PaiementIrDto, PaiementIrCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/paiementIr/');
    }

    public constrcutDto(): PaiementIrDto {
        return new PaiementIrDto();
    }

    public constrcutCriteria(): PaiementIrCriteria {
        return new PaiementIrCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/commun/PaiementIrAdminService.java
package ma.zs.univ.service.facade.admin.commun;

import java.util.List;
import ma.zs.univ.bean.core.commun.PaiementIr;
import ma.zs.univ.dao.criteria.core.commun.PaiementIrCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface PaiementIrAdminService extends IService<PaiementIr,PaiementIrCriteria>  {

    List<PaiementIr> findBySocieteId(Long id);
    int deleteBySocieteId(Long id);
    long countBySocieteId(Long id);




}
